### ISSUE TITLE
Props passed to `<Link>` lose reactivity

### ISSUE THREAD

**Issue by Papooch** [gh0st-work/solid-js-router#3]
When I had an accessor passed to a prop of `Link`, I noticed that the class was not toggled properly when the `isConnected` signal changed.
```tsx
<Link
    href={"/network"}
    classList={{ disabled: !props.isConnected()  }}
>
    Network Game
</Link>
```
This was fixed by wrapping the `Link` element in a span.
```tsx
<span classList={{ disabled: !props.isConnected()  }}>
    <Link href={"/network"}>
        Network Game
    </Link>
</span>
```

I don't know if it happens with the other components as well, but I don't think there's ever a need for reactive props on them.

But this most likely happens because of the [spreading of props](https://github.com/gh0st-work/solid-js-router/blob/46662469a43fb2ad3fb63b14224223ded837008c/src/Link.jsx#L19).

**Comment by gh0st-work** [comment]
I've always just provided calculated memo in props, i.e.
```
const linkClass = createMemo(() => (props.isConnected ? 'disabled' : ''))

...

class={linkClass()}
```

And, yeah, if you want to change "disabled", write it in disabled prop, not in class/className, ofc)


If it can be fixed with props splitting, I'll try in a couple of days. I never even thought that it could affect)

**Comment by gh0st-work** [comment]
Resolved in [v1.4.5](https://github.com/gh0st-work/solid-js-router/releases/tag/1.4.5)

Update the package with npm command:
`npm i @gh0st-work/solid-js-router@^1.4.5`
